apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -37,16 +42,29 @@ class PageTemplateModal extends Component {
 		if ( this.state.isOpen ) {
 			trackView( this.props.segment.id, this.props.vertical.id );
 		}
+
+		// Populate blocks state field, parsing the raw content for each template.
+		const blocks = {};
+		for ( const slug in this.props.templates ) {
+			const template = this.props.templates[ slug ];
+			blocks[ slug ] = template.content ? parseBlocks( template.content ) : [];
+		}
+
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState( { blocks } );
+
 		// eslint-disable-next-line no-console
 		console.timeEnd( 'PageTemplateModal' );
 	}
 
-	setTemplate = ( slug, title, previewBlocks ) => {
+	setTemplate = ( slug, title ) => {
 		this.setState( { isOpen: false } );
 		trackSelection( this.props.segment.id, this.props.vertical.id, slug );
 
 		this.props.saveTemplateChoice( slug );
 
+		const previewBlocks = this.state.blocks[ slug ];
+
 		// Skip inserting if there's nothing to insert.
 		if ( ! previewBlocks || previewBlocks.length === 0 ) {
```

Comment:
Minor: do we need explicit `0` check here? Why not `!previewBlocks.length`?

Reply:
I didn't write this code. Glad to update it.